tools/clang/include/clang/Basic/Attr.td

```diff
@@ -992,6 +992,12 @@ def HLSLNodeTrackRWInputSharing : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+def HLSLTypedResource : InheritableAttr {
+  let Spellings = []; // No spellings!
+  let Subjects = SubjectList<[CXXRecord]>;
+  let Documentation = [Undocumented];
```

Comment:
While this minimally supports the use case at the moment, I think it would be far better to have a more general resource attribute that captures at least `DXIL::ResourceClass`, `DXIL::ResourceKind`, and `DXIL::SamplerKind`, which could be used for all resource objects.  There are already useful inline helper functions in `DxilConstants.h` for checking subsets of resource kinds like `IsTyped`.

Reply:
> I don't think they should share an attribute with other resources.

That's fine by me.